tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -130,6 +127,59 @@ def __init__(self, status, message):
         self.message = message
 
 
+class TimedRunner(object):
+    def __init__(self, logger, func, protocol, url, timeout, extra_timeout):
+        self.func = func
+        self.result = None
+        self.protocol = protocol
+        self.url = url
+        self.timeout = timeout
+        self.extra_timeout = extra_timeout
+        self.result_flag = threading.Event()
+
+    def run(self):
+        if self.set_timeout() is Stop:
+            return Stop
+
+        if self.before_run() is Stop:
+            return Stop
+
+        executor = threading.Thread(target=self.run_func)
+        executor.start()
+
+        flag = self.result_flag.wait(self.timeout + 2 * self.extra_timeout)
```

Comment:
Nit: perhaps call this `finished` or something instead of the generic `flag`.

Reply:
Also maybe worth explaining here why we have a multiplier of 2 (I assume it's because the inner `TestExecutor` uses single `extra_timeout`).